### PR TITLE
[css-transitions] `transition` property should use shortest serialization rule

### PR DESCRIPTION
**Pull request by graouts**
#### 3dee8c9a649aa8cc7d686907b21a6afc6f3132de
<pre>
[css-transitions] `transition` property should use shortest serialization rule
<a href="https://bugs.webkit.org/show_bug.cgi?id=266234">https://bugs.webkit.org/show_bug.cgi?id=266234</a>

Reviewed by Tim Nguyen.

As discussed in <a href="https://github.com/web-platform-tests/wpt/issues/43574">https://github.com/web-platform-tests/wpt/issues/43574</a>, serializing
the `transition` shorthand should always strive to produce the shortest serialization
and omit initial values.

We update the relevant tests, including WPT, to match this expectation and add a new test
for the case where `transition-delay` is not the initial value while `transition-duration`
is since those two properties have the same `&lt;time&gt;` type and `transition-duration` serializes
first.

* LayoutTests/fast/css/shorthand-mismatched-list-crash-expected.txt:
* LayoutTests/fast/css/shorthand-mismatched-list-crash.html:
* LayoutTests/fast/css/transform-inline-style-expected.txt:
* LayoutTests/fast/css/transform-inline-style-remove-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/first-letter-allowed-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-pseudo/first-line-allowed-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-transitions/parsing/transition-behavior.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-transitions/parsing/transition-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-transitions/parsing/transition-computed.html:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::addValueForAnimationPropertyToList):
(WebCore::animationShorthandValue):
(WebCore::singleTransitionValue):
(WebCore::transitionShorthandValue):
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serializeLayered const):
* Source/WebCore/platform/animation/Animation.cpp:
(WebCore::Animation::animationsMatch const):
* Source/WebCore/platform/animation/Animation.h:
(WebCore::Animation::TransitionProperty::operator== const):

Canonical link: <a href="https://commits.webkit.org/272513@main">https://commits.webkit.org/272513@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/9b96947838ae45893eb9c865a52b795e3d7ff96b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30132 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8797 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31836 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32636 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27247 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30807 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10997 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6042 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27257 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30432 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7382 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/26983 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6310 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6463 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27017 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33983 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27475 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27514 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32652 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6410 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4577 "Found 1 new test failure: fullscreen/full-screen-layer-dump.html (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30458 "Found 1 new API test failure: /WebKitGTK/TestResources:/webkit/WebKitWebView/sync-request-on-max-conns (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8160 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/26771 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7463 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7164 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6939 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/css/ComputedStyleExtractor.cpp
Comment (nt1m) [inline Source/WebCore/css/ComputedStyleExtractor.cpp:1689]: Can we assert that the list isn't empty?

**Review by nt1m — APPROVED** [review]

File: Source/WebCore/css/ComputedStyleExtractor.cpp
Comment (graouts) [reply Source/WebCore/css/ComputedStyleExtractor.cpp:1689]: The `CSSValueListBuilder` or resulting `CSSValueList`? I guess both will have the same number of items.

Comment (nt1m) [reply Source/WebCore/css/ComputedStyleExtractor.cpp:1689]: CSSValueListBuilder

Comment (nt1m) [reply Source/WebCore/css/ComputedStyleExtractor.cpp:1689]: but either is fine

Comment (graouts) [reply Source/WebCore/css/ComputedStyleExtractor.cpp:1689]: This review feedback got me to think about `singleTransitionValue()` yielding an empty list for an `all` value and sure enough it did, so that's a bug to fix and additional test to write. Will file a new PR with those changes and the suggested assertion.

**Comment by graouts** [comment]
Will hold off for a bit before landing this so that we have a bit more time to determine whether there are compatibility issues.

File: Source/WebCore/css/ComputedStyleExtractor.cpp
Comment (graouts) [reply Source/WebCore/css/ComputedStyleExtractor.cpp:1689]: The `all` case was fixed in a subsequent version of this patch, and tested.

**Pull request merged**
